### PR TITLE
One-way dependency

### PR DESCRIPTION
**Pull request by willu47**
[Finishes #142147373](https://www.pivotaltracker.com/story/show/142147373)

Also [Finishes #140395691](https://www.pivotaltracker.com/story/show/140395691)

## Implements a single one-way dependency
- Takes the output of one model and converts it for input to a second model

## Development
- Removes `from_model` attribute of inputs
- Harmonizes implemention of namedtuple used in smif.inputs and smif.outputs

## API Change
inputs and outputs are now defined as a list of parameters, rather than a dict of parameter types containing a list of parameters

### CODE REVIEW

File: smif/sector_model.py [hunk 1]
```diff
@@ -49,8 +48,8 @@ def __init__(self):
 
         self.interventions = []
 
-        self._inputs = ModelInputs({})
-        self._outputs = ModelOutputs({})
+        self._inputs = ModelParameters({})
```
Comment (tomalrussell) [inline smif/sector_model.py]: Great to collapse Inputs/Outputs.
Should that dict be a list?

Comment (willu47) [inline smif/sector_model.py]: Yep, good catch

File: smif/sos_model.py [hunk 2]
```diff
@@ -611,19 +705,22 @@ def _check_dependencies(self):
...
+                    msg = "Missing dependency: {} depends on {}, " + \
+                        "which is not supplied."
+                    raise AssertionError(msg.format(model_name, dep.name))
+
+                for source in providers:
+                    if source == 'scenario':
+                        continue
 
-                dependency_graph.add_edge(model_name, dep.from_model)
+                    dependency_graph.add_edge(model_name, source)
```
Comment (tomalrussell) [inline smif/sos_model.py]: :+1:

File: smif/parameters.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+    The ``name`` entry denotes the unique identifier of a model or scenario
+    parameter.
+    The ``spatial_resolution`` and ``temporal_resolution`` are references to the
+    catalogues held by the :class:`~smif.sector_model.SosModel` which define the
+    available conversion formats.
+
+    Parameters
+    ----------
+    parameters: dict
```
Comment (willu47) [inline smif/parameters.py]: Thanks - just tidying docstrings now.

**Comment by willu47** [comment]
@tomalrussell I think `smif.parameters.ParameterList` can be squashed into `smif.parameters.ModelParameters` but am not sure how...

**Pull request merged**